plugins/hydra_optuna_sweeper/tests/test_optuna_sweeper_plugin.py

```diff
@@ -0,0 +1,122 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+from pathlib import Path
+from typing import Any
+
+import pytest
+from hydra.core.override_parser.overrides_parser import OverridesParser
+from hydra.core.plugins import Plugins
+from hydra.plugins.sweeper import Sweeper
+from hydra.test_utils.test_utils import TSweepRunner, chdir_plugin_root, get_run_output
+from omegaconf import DictConfig, OmegaConf
+from optuna.distributions import (
+    CategoricalDistribution,
+    DiscreteUniformDistribution,
+    IntLogUniformDistribution,
+    IntUniformDistribution,
+    LogUniformDistribution,
+    UniformDistribution,
+)
+
+from hydra_plugins.hydra_optuna_sweeper import _impl
```

Comment:
Ahh I am not sure here but iirc all python files (whose name doesnt start with `_`) are loaded during hydra init. This can slowdown hydra startup. Thats why we write the core logic (interacting with say Optuna) in `_impl.py`. `tests` dir may be ignored. @omry can confirm.

 Alternatively, we can move this import inside the function we need it.

Reply:
> `tests` dir may be ignored.

I agree with you. IMHO, the test files are usually not loaded when we import libraries. The test of [`hydra_nevergrad_sweeper`](https://github.com/facebookresearch/hydra/blob/599205ffa771045429a6a32ef69a464602d31e15/plugins/hydra_nevergrad_sweeper/tests/test_nevergrad_sweeper_plugin.py#L13) also import `_impl.py`, and I think we can keep this line as it is.